src/librbd/cache/ImageCache.h

```diff
@@ -30,7 +37,7 @@ struct ImageCache {
   virtual void aio_discard(uint64_t offset, uint64_t length,
                            uint32_t discard_granularity_bytes,
                            Context *on_finish) = 0;
-  virtual void aio_flush(Context *on_finish) = 0;
+  virtual void aio_flush(Context *on_finish, io::FlushSource flush_source = io::FLUSH_SOURCE_USER) = 0;
```

Comment:
Nit: avoid defaulting parameters, just fix the code breakages. Also try to keep the ```on_finish``` as the last parameter.

Reply:
How do you feel about alternate prototypes with fewer args so we don't have to change every call to a method that gains an arg? I'm not sure why I didn't do that here, actually.

Reply:
That basically sounds the same as defaulting parameters.

Reply:
So use 'virtual void aio_flush(Context *on_finish, io::FlushSource flush_source) = 0;' to replace the existed declaration, and fix existed calls.